docs/node/mainnet/e-upgrade.md

```diff
@@ -100,3 +100,299 @@ upgrade. For now, please take a look at the related [Testnet upgrade].
 :::
 
 [Testnet upgrade]: ../testnet/upgrade-log.md
+
+### Voting
+
+On XXX, an upgrade proposal will be proposed which (if accepted) will schedule
+the upgrade on epoch **XXX**.
+
+For an optimized voting experience, we recommend utilizing the [Oasis CLI].
+Follow these steps to cast your vote:
+1. [Import your keys into the wallet]
+2. [Cast your vote]
+
+### Upgrade Instructions
+
+The following steps should be performed only after the network has reached the
+upgrade epoch and has halted:
+
+1. Wait for the network to reach the upgrade epoch and halt. The node
+    will automatically stop without any action required on your part. After the
+    network has halted, proceed to the next steps.
+
+2. Download the Mainnet genesis file published in the
+   [Mainnet XXX release].
+
+:::info
+
+Mainnet state at epoch **XXX** will be exported and migrated to a 23.0
+compatible genesis file.
+
+The new genesis file will be published on the above link soon after reaching the
+upgrade epoch.
+
+:::
+
+3. Verify the provided Mainnet upgrade genesis file by comparing it to the
+   local network state dump.
+
+   Find the `genesis-mainnet-XXX.json` file in the `exports` subdirectory in your data dir (e.g. `/node/`, `/srv/oasis/node/`)  and run
+   sha256sum on it. Afterwards, compare it with the hash that we will share on
+   the `#node-operators` Discord channel.
+
+   The state changes are described in the [State Changes](#state-changes)
+   section below.
+
+4. Replace the old genesis file with the new Mainnet genesis file.
+
+5. Ensure your node will remain stopped by disabling auto-starting via your
+   process manager (e.g., [systemd] or [Supervisor])
+
+6. Back up the entire data directory of your node. Verify that the backup
+includes the following folders:
+- for consensus: `tendermint/abci-state` and `tendermint/data`
+- for runtimes: `runtimes/*/mkvs_storage.badger.db` and
+`runtimes/*/worker-local-storage.badger.db`
+
+7. [Wipe state]. This must be performed _before_ replacing the Oasis Node
+   binary. In case you are upgrading ParaTimes/runtimes ensure you read the
+   following Danger section:
+
+:::danger
+
+State of ParaTimes/runtimes is not affected by this upgrade and MUST NOT be
+wiped. Wiping state for confidential ParaTimes will prevent your compute or
+key manager node from transitioning to the new network.
+
+Make sure to include the `--preserve.mkvs_database` flag if you are wiping
+blockchain state on a runtime node and need to preserve runtime state.
```

Comment:
This is not needed as this flag is the default.

Reply:
We had a lot of questions about this flag. Also https://docs.oasis.io/node/run-your-node/maintenance/wiping-node-state#state-wipe-and-keep-node-identity does not look like it is default?

Reply:
It sure is:
```
$ ./bin/oasis-node unsafe-reset --help
reset the node state (UNSAFE)

Usage:
  oasis-node unsafe-reset [flags]

Flags:
  -n, --dry_run                  don't actually do anything, just show what will be done
      --force                    force
  -h, --help                     help for unsafe-reset
      --preserve.local_storage   preserve per-runtime untrusted local storage (default true)
      --preserve.mkvs_database   preserve per-runtime MKVS database (default true)
...
```

Reply:
The linked doc seems out of date, this has been the default for a long time now. The doc should be fixed instead.

Reply:
Will edit the linked docs after we finish this docs.

Reply:
Oh something crashed. Let me check what it is